File: test/fixtures/javascript/simple.js
// Expected complexity: 0
function empty() {
  return 42;
}

// Expected complexity: 1
function singleIf(a) {
  if (a) {       // +1
    return true;
  }
  return false;
}

// Expected complexity: 4
function ifElseIfElse(a, b) {
  if (a) {          // +1
    return 1;
  } else if (b) {   // +1
    return 2;
  } else {           // +1
    return 3;
  }
}

// Expected complexity: 3
function simpleLoop(arr) {
  for (let i = 0; i < arr.length; i++) {  // +1
    if (arr[i] > 0) {                      // +2 (nesting=1)
      continue;
    }
  }
}

// Expected complexity: 1
function booleanAnd(a, b) {
  return a && b;  // +1 (&&)
}

// Expected complexity: 2
function mixedBooleans(a, b, c) {
  return a && b || c;  // +1 (&&) +1 (||)
}

// Expected complexity: 1
function sameBooleans(a, b, c) {
  return a && b && c;  // +1 (&&) - same operator, only counted once
}

File: test/fixtures/python/simple.py
# Expected complexity: 0
def empty():
    return 42

# Expected complexity: 1
def single_if(a):
    if a:       # +1
        return True
    return False

# Expected complexity: 3
def if_elif_else(a, b):
    if a:           # +1
        return 1
    elif b:         # +1
        return 2
    else:           # +1
        return 3

# Expected complexity: 3
def simple_loop(arr):
    for item in arr:        # +1
        if item > 0:        # +2 (nesting=1)
            continue

# Expected complexity: 1
def boolean_and(a, b):
    return a and b  # +1 (and)

# Expected complexity: 2
def mixed_booleans(a, b, c):
    return a and b or c  # +1 (and) +1 (or)
